Source/WebCore/accessibility/AccessibilityRenderObject.cpp

```diff
@@ -1078,6 +1078,26 @@ static bool webAreaIsPresentational(RenderObject* renderer)
     return ownerElement && nodeHasPresentationRole(*ownerElement);
 }
 
+bool AccessibilityRenderObject::shouldWebAreaExposeChildren() const
+{
+    if (!isWebArea())
+        return true;
+
+    CheckedPtr renderView = dynamicDowncast<RenderView>(m_renderer.get());
+    CheckedPtr frameRenderer = renderView ? renderView->frameView().frame().ownerRenderer() : nullptr;
+
+    while (frameRenderer) {
+        const auto usedVisibility = frameRenderer->style().usedVisibility();
+        if (usedVisibility != Visibility::Visible || frameRenderer->style().effectiveInert())
+            return false;
+
+        renderView = frameRenderer->document().renderView();
+        frameRenderer = renderView ? renderView->frameView().frame().ownerRenderer() : nullptr;
+    }
+
+    return true;
+}
```

Comment:
We are already doing this type of check in

AccessibilityObjectInclusion AccessibilityObject::defaultObjectInclusion() const
{
if (auto* style = this->style()) {
if (style->effectiveInert())
return AccessibilityObjectInclusion::IgnoreObject;
if (style->usedVisibility() != Visibility::Visible) {
// aria-hidden is meant to override visibility as the determinant in AX hierarchy inclusion.
if (equalLettersIgnoringASCIICase(getAttribute(aria_hiddenAttr), "false"_s))
return AccessibilityObjectInclusion::DefaultBehavior;

return AccessibilityObjectInclusion::IgnoreObject;
}
}

Do we need to do it here again?

Reply:
Inside iframes, `effectiveInert()` and `usedVisibility()` won't return the properties if the visibility/inert is set in a parent outside the iframe. That's why we need to do the walkup of the documents/renderers here.